File: backend/src/main/java/com/englishlearning/backend/service/StudentVocabularyService.java
package com.englishlearning.backend.service;

import com.englishlearning.backend.dto.request.SaveVocabularyRequest;
import com.englishlearning.backend.dto.response.SavedVocabularyResponse;
import com.englishlearning.backend.enums.LearningStatus;

import java.util.List;

public interface StudentVocabularyService {

    SavedVocabularyResponse saveVocabulary(
            Long userId,
            SaveVocabularyRequest request
    );

    List<SavedVocabularyResponse> getAll(
            Long userId
    );

    List<SavedVocabularyResponse> getByStatus(
            Long userId,
            LearningStatus status
    );

    SavedVocabularyResponse updateStatus(
            Long userId,
            Long studentVocabularyId,
            LearningStatus status
    );
}

File: frontend/src/services/studentVocabularyService.js
import axiosClient from "../api/axiosClient";

const studentVocabularyService = {
  saveVocabulary(data) {
    return axiosClient.post("/v1/student-vocabularies", data);
  },

  getAll() {
    return axiosClient.get("/v1/student-vocabularies");
  },
  getByStatus(status) {
    return axiosClient.get("/v1/student-vocabularies", {
      params: {
        status,
      },
    });
  },

  updateStatus(studentVocabularyId, status) {
    return axiosClient.patch(
      `/v1/student-vocabularies/${studentVocabularyId}/status`,
      {
        status,
      },
    );
  },
};

export default studentVocabularyService;
